utf8proc.c

```diff
@@ -186,7 +188,38 @@ UTF8PROC_DLLEXPORT utf8proc_ssize_t utf8proc_encode_char(utf8proc_int32_t uc, ut
 }
 
 /* internal "unsafe" version that does not check whether uc is in range */
-static const utf8proc_property_t *get_property(utf8proc_int32_t uc) {
+utf8proc_ssize_t utf8proc_unsafe_encode_char(utf8proc_int32_t uc, utf8proc_uint8_t *dst) {
+   if (uc < 0x00) {
+      return 0;
+   } else if (uc < 0x80) {
+      dst[0] = uc;
+      return 1;
+   } else if (uc < 0x800) {
+      dst[0] = 0xC0 + (uc >> 6);
+      dst[1] = 0x80 + (uc & 0x3F);
+      return 2;
+   } else if (uc == 0xFFFF) {
+       dst[0] = 0xFF;
+       return 1;
+   } else if (uc == 0xFFFE) {
+       dst[0] = 0xFE;
+       return 1;
+   } else if (uc < 0x10000) {
+      dst[0] = 0xE0 + (uc >> 12);
+      dst[1] = 0x80 + ((uc >> 6) & 0x3F);
+      dst[2] = 0x80 + (uc & 0x3F);
+      return 3;
+   } else if (uc < 0x110000) {
+      dst[0] = 0xF0 + (uc >> 18);
+      dst[1] = 0x80 + ((uc >> 12) & 0x3F);
+      dst[2] = 0x80 + ((uc >> 6) & 0x3F);
+      dst[3] = 0x80 + (uc & 0x3F);
+      return 4;
+   } else return 0;
+}
+
+/* internal "unsafe" version that does not check whether uc is in range */
+static const utf8proc_property_t *utf8proc_unsafe_get_property(utf8proc_int32_t uc) {
```

Comment:
I added those because I mistakenly thought that's what you were asking me to do... I'll go back and  put those back... (this is what you get for programming on 3 days of only 2 hours of sleep!)